account_payment_return/models/account_move.py

```diff
@@ -33,6 +33,7 @@ def prepare_values_returned_widget(self, line_id, amount, is_return=False):
             payment_method_name = line_id.payment_method_line_id.name
         except AttributeError:
             payment_method_name = False
+        ref = line_id.memo if hasattr(line_id, "memo") else line_id.ref
```

Comment:
```suggestion
        ref = line_id.memo if hasattr(line_id, "memo") and line_id.memo else line_id.ref
```
If `memo` is empty, let's fill with `ref`.

Reply:
I don't think that will work, since if `line_id` has the `memo` field (so `hasattr(line_id, "memo")` returns True), the record won't have the `ref` field. If we try to fill it with `ref` we might get an error.

Reply:
> if `line_id` has the `memo` field, the record won't have the `ref` field

Why?

> If we try to fill it with `ref` we might get an error.

Why?


(I don't know, I am just asking to know it better)

Reply:
Check the initial comment in the PR with the explanation: https://github.com/OCA/account-payment/pull/882#issue-3444296421

Reply:
Because the `line_id` parameter in this method can be, either a record of the `account.payment` model, or a record of the `account.move.line` model.

If `line_id` is an `account.payment`, then it will have the `memo` field, and id won't have the `ref` field. That's why trying to get the `ref` field in that case will throw an error, since the attribute does not exist for that record or model.

However, if `line_id` is an `account.move.line`, then it won't have the `memo` field, but it will have `ref`. So that covers the other scenario.

Hope it is clear now!! :)